docs/README.md

```
@@ -0,0 +1,75 @@
+This documentation follows the [Diataxis framework](https://diataxis.fr/), which organizes
+content into four distinct types based on user needs. Use these guidelines when writing or
+reviewing documentation.
+
+## Folder Structure
+
+| Location | Diataxis Type | Purpose |
+|--------|---------------|---------|
+| `tutorials/` | Tutorials | Learning-oriented lessons |
+| `concepts/` | Explanation | Understanding-oriented background |
+| `guides/` | How-to Guides | Task-oriented problem solving |
+| Rustdoc (generated) | Reference | Information-oriented technical descriptions |
```

markdownlint-cli2 (0.18.1)

Tables should be surrounded by blank lines

(MD058, blanks-around-tables)